pkg/blockbuilder/tsdb_test.go

```diff
@@ -405,111 +399,42 @@ func mockUploaderFunc(t *testing.T, destDir string) blockUploader {
 // It is important that processing empty request is a success, as in says all samples were processed,
 // so that checkpointing can be done correctly.
 func TestProcessingEmptyRequest(t *testing.T) {
+	partitionID := int32(0)
 	userID := "1"
 
 	overrides := validation.NewOverrides(defaultLimitsTestConfig(), nil)
 	metrics := newTSDBBBuilderMetrics(prometheus.NewPedanticRegistry())
-	builder := NewTSDBBuilder(log.NewNopLogger(), t.TempDir(), mimir_tsdb.BlocksStorageConfig{}, overrides, metrics, 0)
+	builder := NewTSDBBuilder(log.NewNopLogger(), t.TempDir(), partitionID, mimir_tsdb.BlocksStorageConfig{}, overrides, metrics, 0)
+
+	ctx := user.InjectOrgID(t.Context(), userID)
 
 	// Has a timeseries with no samples.
-	var rec kgo.Record
-	rec.Key = []byte(userID)
-	req := mimirpb.WriteRequest{}
-	req.Timeseries = append(req.Timeseries, mimirpb.PreallocTimeseries{
+	req1 := mimirpb.WriteRequest{}
+	req1.Timeseries = append(req1.Timeseries, mimirpb.PreallocTimeseries{
 		TimeSeries: &mimirpb.TimeSeries{
 			Labels:  []mimirpb.LabelAdapter{{Name: "foo", Value: "bar"}},
 			Samples: []mimirpb.Sample{},
 		},
 	})
-	data, err := req.Marshal()
-	require.NoError(t, err)
-	rec.Value = data
-	err = builder.Process(context.Background(), &rec)
+	err := builder.PushToStorageAndReleaseRequest(ctx, &req1)
 	require.NoError(t, err)
 
 	// Has no timeseries.
-	req.Timeseries = req.Timeseries[:0]
-	data, err = req.Marshal()
-	require.NoError(t, err)
-	rec.Value = data
-	err = builder.Process(context.Background(), &rec)
+	req2 := mimirpb.WriteRequest{}
+	req2.Timeseries = make([]mimirpb.PreallocTimeseries, 0)
+	err = builder.PushToStorageAndReleaseRequest(ctx, &req2)
 	require.NoError(t, err)
 
 	require.NoError(t, builder.tsdbs[tsdbTenant{0, userID}].Close())
 }
 
-func TestTSDBBuilder_KafkaRecordVersion(t *testing.T) {
```

Comment:
Note to a reviewer: I had to rework this test completely because, with the changes in this PR, the parts about `KafkaRecordVersion` are hidden from the block-builder.

Reply:
I've removed the test completely 👍